src/main/app/src/components/haku/Sivutus.js

```diff
@@ -0,0 +1,153 @@
+import React, { Component } from 'react';
+import qs from 'query-string';
+import '../../assets/css/hakutulos.css'
+import {observer, inject} from 'mobx-react';
+
+@inject("hakuStore")
+@inject("urlStore")
+@observer
+class Sivutus extends Component {
+
+    componentDidMount() {
+        //this.handleRefresh();
+    }
+
+    componentWillReceiveProps(nextProps) {
+        //this.props = nextProps;
+        //this.handleHistory()
+    }
+
+    handleHistory() {
+        const queryParamToggle = qs.parse(this.props.location.search).toggle;
+        if(this.props.hakuStore.keyword !== this.props.match.params.keyword) {
+            this.props.hakuStore.keyword = this.props.match.params.keyword;
+            this.search(queryParamToggle);
+        } else {
+            this.props.hakuStore.toggleKoulutus = ('oppilaitos' !== queryParamToggle)
+        }
+    }
+
+    changeUrl() {
+        if(this.props.hakuStore.keyword === this.props.match.params.keyword) {
+            this.props.history.replace(this.props.hakuStore.createHakuUrl);
+        } else {
+            this.props.history.push(this.props.hakuStore.createHakuUrl);
+        }
+    }
+
+    searchAction(newKeyword) {
```

Comment:
Käytetäänkö näitä handleHistory-, changeUrl- ja searchAction-metodeita johonkin? Ihan kuin ne olisi vaan copy-pastettu tähän Haku-komponentista. Ne voisi poistaa.